modules/dns_common/dns_common.rb

```diff
@@ -24,5 +27,43 @@ def dns_find key
     rescue Resolv::ResolvError
       false
     end
+
+    def ptr_to_ip ptr
+     if ptr =~ /\.in-addr\.arpa$/
+       ptr.split('.')[0..-3].reverse.join('.')
+     elsif ptr =~ /\.ip6\.arpa$/
+       ptr.split('.')[0..-3].reverse.each_slice(4).inject([]) {|address, word| address << word.join}.join(":")
+     else
+       raise Proxy::Dns::Error.new("Not a PTR record: '#{ptr}'")
+     end
+    end
+
+    # conflict methods return values:
+    # no conflict: -1; conflict: 1, conflict but record / ip matches: 0
+    def a_record_conflicts(fqdn, ip)
+      if ip_addr = validate_ip(ip)
+        addresses = resolver.getaddresses(fqdn)
+        return -1 if addresses.empty?
+        addresses.each { |address| return 0 if IPAddr.new(address.to_s).to_s.casecmp(ip_addr.to_s) == 0 }
```

Comment:
You can compare IPAddr instances directly, no need to convert them to strings. Also, use #any? instead of each {... predicate_here }:

```
return 0 if addresses.any? {|a| IPAddr.new(a) == ip_addr}
```